controllers/appgroup_controller_test.go

```diff
@@ -120,6 +119,77 @@ var _ = Describe("ApplicationGroup Controller", func() {
 
 		})
 
+		It("Should create create only application releases with subchart nil successfully", func() {
+			applicationGroup := defaultAppGroup(name)
+			applicationGroup.Name = name
+			applicationGroup.Namespace = DefaultNamespace
+			for i := range applicationGroup.Spec.Applications {
+				applicationGroup.Spec.Applications[i].Spec.Subcharts = nil
+			}
+			key := client.ObjectKeyFromObject(applicationGroup)
+
+			By("Applying the bookinfo object to the cluster")
+			err := k8sClient.Create(ctx, applicationGroup)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Defer the cleanup so that we delete the appGroup after creation
+			defer func() {
+				By("Deleting the bookinfo object from the cluster")
+				patch := client.MergeFrom(applicationGroup.DeepCopy())
+				controllerutil.RemoveFinalizer(applicationGroup, v1alpha1.AppGroupFinalizer)
+				_ = k8sClient.Patch(ctx, applicationGroup, patch)
+				_ = k8sClient.Delete(ctx, applicationGroup)
+			}()
+
+			helmReleaseList := &fluxhelmv2beta1.HelmReleaseList{}
+			err = k8sClient.List(ctx, helmReleaseList, client.InNamespace(name))
+			Expect(err).ToNot(HaveOccurred())
+			oldHelmReleaseCount := len(helmReleaseList.Items)
+
+			By("Making sure that the workflow goes into a running state")
+			Eventually(func() bool {
+				workflow := &v1alpha12.Workflow{}
+				workflowKey := types.NamespacedName{Name: applicationGroup.Name, Namespace: applicationGroup.Namespace}
+				_ = k8sClient.Get(ctx, workflowKey, workflow)
+				return workflow.Status.Phase == v1alpha12.NodeRunning
+			}, time.Minute, time.Second).Should(BeTrue())
+
+			By("Waiting for the bookinfo object to reach a succeeded reason")
+			Eventually(func() bool {
+				applicationGroup = &v1alpha1.ApplicationGroup{}
+				if err := k8sClient.Get(ctx, key, applicationGroup); err != nil {
+					return false
+				}
+				return applicationGroup.GetReadyCondition() == meta.SucceededReason
+			}, DefaultTimeout, time.Second).Should(BeTrue())
+
+			By("checking that the all the HelmReleases have come up and are in a ready state")
+			err = k8sClient.List(ctx, helmReleaseList, client.InNamespace(name))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(helmReleaseList.Items)).To(Equal(oldHelmReleaseCount + 2))
+			allReady := true
+			for _, release := range helmReleaseList.Items {
+				if condition := meta.GetResourceCondition(&release, meta.ReadyCondition); condition.Reason == meta.SucceededReason {
+					allReady = false
+				}
+			}
+			Expect(allReady).To(BeTrue())
+
+			// Wait for all the HelmReleases to delete
+			err = k8sClient.Delete(ctx, applicationGroup)
```

Comment:
Do we need to wait for deletion for this test case?

Reply:
If I force through the `Delete` any chance it would interfere with the other tests in the suite? If not then I can remove this 👍

Reply:
Test automatically does cleanup so no need to even call delete